coderd/workspaces_test.go

```diff
@@ -426,47 +426,441 @@ func TestWorkspace(t *testing.T) {
+							{Name: "param1", Value: "value1"},
+							{Name: "param2", Value: "value2"},
+						},
+					},
+					{Name: "test2"},
+					{Name: "test3"},
+				},
+				templateVersionParameters: []*proto.RichParameter{
+					{Name: "param1", Type: "string", Required: true},
+					{Name: "param2", Type: "string", Required: true},
+				},
+				selectedPresetIndex: 0,
+			},
+			{
+				name: "Multiple Presets - Middle Has Parameters",
+				presets: []*proto.Preset{
+					{Name: "test1"},
+					{
+						Name: "test2",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param1", Value: "value1"},
+							{Name: "param2", Value: "value2"},
+						},
+					},
+					{Name: "test3"},
+				},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       1,
+			},
+			{
+				name: "Multiple Presets - Middle Has Matching Parameters",
+				presets: []*proto.Preset{
+					{Name: "test1"},
+					{
+						Name: "test2",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param1", Value: "value1"},
+							{Name: "param2", Value: "value2"},
+						},
+					},
+					{Name: "test3"},
+				},
+				templateVersionParameters: []*proto.RichParameter{
+					{Name: "param1", Type: "string", Required: true},
+					{Name: "param2", Type: "string", Required: true},
+				},
+				selectedPresetIndex: 1,
+			},
+			{
+				name: "Multiple Presets - Last Has Parameters",
+				presets: []*proto.Preset{
+					{Name: "test1"},
+					{Name: "test2"},
+					{
+						Name: "test3",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param1", Value: "value1"},
+							{Name: "param2", Value: "value2"},
+						},
+					},
+				},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       2,
+			},
+			{
+				name: "Multiple Presets - Last Has Matching Parameters",
+				presets: []*proto.Preset{
+					{Name: "test1"},
+					{Name: "test2"},
+					{
+						Name: "test3",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param1", Value: "value1"},
+							{Name: "param2", Value: "value2"},
+						},
+					},
+				},
+				templateVersionParameters: []*proto.RichParameter{
+					{Name: "param1", Type: "string", Required: true},
+					{Name: "param2", Type: "string", Required: true},
+				},
+				selectedPresetIndex: 2,
+			},
+			{
+				name: "Multiple Presets - All Have Parameters",
+				presets: []*proto.Preset{
+					{
+						Name: "test1",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param1", Value: "value1"},
+						},
+					},
+					{
+						Name: "test2",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param2", Value: "value2"},
+						},
+					},
+					{
+						Name: "test3",
+						Parameters: []*proto.PresetParameter{
+							{Name: "param3", Value: "value3"},
+						},
+					},
+				},
+				templateVersionParameters: []*proto.RichParameter{},
+				selectedPresetIndex:       1,
+			},
+			{
+				name: "Multiple Presets - All Have Matching Parameters",
```

Comment:
I think a case that's missing here is two presets match the same parameters.

Reply:
added